org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/MavenCapability.java

```diff
@@ -85,13 +85,7 @@ public static MavenCapability createMavenParent(ArtifactKey key) {
   }
 
   static String getId(ArtifactKey key) {
-    StringBuilder sb = new StringBuilder();
-    sb.append(key.groupId());
-    sb.append(':').append(key.artifactId());
-    if(key.classifier() != null) {
-      sb.append(':').append(key.classifier());
-    }
-    return sb.toString();
+    return key.groupId() + ":" + key.artifactId() + (key.classifier() != null ? ":" + key.classifier() : "");
```

Comment:
I find this hard to read to have concatenation and trigraph mixed in one expression. Objects.RequireNonNullOrElse might be more expressive.